tecken/symbolicate/views.py

```diff
@@ -725,31 +737,41 @@ def load_symbol(self, filename, debug_id):
         for line in stream:
             total_size += len(line)
             line_number += 1
-            if line.startswith('PUBLIC '):
-                fields = line.strip().split(None, 3)
-                if len(fields) < 4:
+
+            try:
+                found, = starts_regex.findall(line)
+                prefix = found[0]
+            except ValueError:
+                # If it didn't match anything at the start, don't bother
+                # with this line.
+                continue
+
+            rest = starts_regex.sub('', line)
+            if prefix == 'PUBLIC':
+                fields = rest.strip().split(None, 2)
+                if len(fields) != 3:
                     logger.warning(
                         'PUBLIC line {} in {} has too few fields'.format(
                             line_number,
                             url,
                         )
                     )
                     continue
-                address = int(fields[1], 16)
-                symbol = fields[3]
+                address = int(fields[0], 16)
+                symbol = fields[2]
                 public_symbols[address] = symbol
-            elif line.startswith('FUNC '):
-                fields = line.strip().split(None, 4)
-                if len(fields) < 4:
+            elif prefix == 'FUNC':
+                fields = rest.strip().split(None, 3)
+                if len(fields) != 4:
                     logger.warning(
                         'FUNC line {} in {} has too few fields'.format(
                             line_number,
                             url,
                         )
                     )
                     continue
-                address = int(fields[1], 16)
-                symbol = fields[4]
+                address = int(fields[0], 16)
+                symbol = fields[3]
                 func_symbols[address] = symbol
 
         # Prioritize PUBLIC symbols over FUNC symbols # XXX why?
```

Comment:
I noticed this comment while looking at the code. Is this actually true? If so this is the opposite of what we actually want (FUNC symbols should take priority over PUBLIC symbols.) If that's the case we should open a new issue to fix that separately.

Reply:
This is a relic from https://github.com/mozilla/Snappy-Symbolication-Server/blob/6a3347dc32d5ecaccaafd3219f341d7b7c2d0cf0/snappy/DiskCache_DiskCache.py#L380-L382 which I just copied (although I call the dict variables differently). So is this wrong? Then we should definitely start a new ticket.